### PR TITLE
WT-10563 Print calculated checksum and expected checksum when we have a checksum mismatch

### PR DESCRIPTION
**Pull request by Sean04**

### CODE REVIEW

File: src/block/block_read.c [hunk 1]
```diff
@@ -215,9 +215,12 @@ __wt_block_read_off(WT_SESSION_IMPL *session, WT_BLOCK *block, WT_ITEM *buf, uin
...
             __wt_errx(session,
               "%s: potential hardware corruption, read checksum error for %" PRIu32
-              "B block at offset %" PRIuMAX
-              ": calculated block checksum doesn't match expected checksum",
-              block->name, size, (uintmax_t)offset);
+              "B block at offset %" PRIuMAX ": calculated block checksum of %#" PRIx32
+              " doesn't match expected checksum of %#" PRIx32,
+              block->name, size, (uintmax_t)offset,
+              __wt_checksum(
+                buf->mem, F_ISSET(&swap, WT_BLOCK_DATA_CKSUM) ? size : WT_BLOCK_COMPRESS_SKIP),
```
Comment (ddanderson) [inline src/block/block_read.c:222]: Can you put `F_ISSET(&swap, WT_BLOCK_DATA_CKSUM) ? size : WT_BLOCK_COMPRESS_SKIP)` into a variable, perhaps `size_t check_size;` ?

**Review by ddanderson — COMMENTED** [review]
The change looks good - but I'd like to see a small clean up.

**Review by sueloverso — APPROVED** [review]
LGTM after Don's comment.

File: src/block/block_read.c (hunk 1 above)
Comment (Sean04) [reply src/block/block_read.c]: Yeah, that would be neater, thanks.

**Comment by Sean04** [comment]
@sueloverso I've added the checksum to the messages when reading the file metadata and when opening the log.

**Review by sueloverso — APPROVED** [review]
LGTM

**Comment by Sean04** [comment]
evergreen merge

**Pull request merged**